lib/sidekiq_adhoc_job/utils/string.rb

```diff
@@ -0,0 +1,77 @@
+# References: https://github.com/hanami/utils/blob/master/lib/hanami/utils/string.rb
+module SidekiqAdhocJob
+  module Utils
+    class String
+
+      EMPTY_STRING               ||= ''.freeze
+      CLASSIFY_SEPARATOR         ||= '_'.freeze
+      NAMESPACE_SEPARATOR        ||= '::'.freeze
+      UNDERSCORE_SEPARATOR       ||= '/'.freeze
+      UNDERSCORE_DIVISION_TARGET ||= '\1_\2'.freeze
+      DASHERIZE_SEPARATOR        ||= '-'.freeze
+      CLASSIFY_WORD_SEPARATOR    ||= /#{CLASSIFY_SEPARATOR}|#{NAMESPACE_SEPARATOR}|#{UNDERSCORE_SEPARATOR}|#{DASHERIZE_SEPARATOR}/.freeze
+
+      def self.classify(input)
+        string = ::String.new(input.to_s)
+        words = underscore(string).split(CLASSIFY_WORD_SEPARATOR).map!(&:capitalize)
+        delimiters = underscore(string).scan(CLASSIFY_WORD_SEPARATOR)
+
+        delimiters.map! do |delimiter|
+          delimiter == CLASSIFY_SEPARATOR ? EMPTY_STRING : NAMESPACE_SEPARATOR
+        end
+
+        words.zip(delimiters).join
+      end
+
+      def self.underscore(input)
```

Comment:
This class is mostly copied from Hanami string util for the purpose of not having dependency on Hanami in this project. I didn't look at the implementation and didn't question the implementation, since I assume their own community would have reviewed and QA-ed.

But I should perhaps have copied their comments as well, if that helps you (or anyone else) understand the code better.

Reply:
I see! I didn't know that it was from Hanami. Got it 👍